src/api/user.ts

```diff
@@ -31,13 +33,27 @@ export const getCurrentAuthenticatedUser = () => {
   return async (dispatch: Dispatch<any>, getState: () => RootStore) => {
     try {
       const cognitoUser = await Auth.currentAuthenticatedUser();
-      const user = await cognitoUser.attributes;
-      dispatch(setUser(user));
+      if (cognitoUser != null) {
+        await cognitoUser.getSession(async (error: any, session: CognitoUserSession) => {
```

Comment:
Har bruken av async/await noen effekt her? Mistenker at dette vil kjøre som vanlig callback uavhengig av async/await, og at riktig bruk av async/await vil nære noe slik som `const session = await cognitouser.getSession()`. Hvis det ikke er så enkelt kan vi bare kutte async/await her

Reply:
det er uansett en async funksjon siden det er en thunk. blir ikke noe kortere kode av å bruke promises. men det du mener er kanskje at det er unødvendig med try/catch?

Reply:
Mistenker at async/await ikke har noen effekt for `getSession` og `getuserAttributes` atm, siden responsen håndteres via callback. I dokumentasjonen har de samme bare uten async/await: https://docs.aws.amazon.com/cognito/latest/developerguide/using-amazon-cognito-user-identity-pools-javascript-examples.html

Reply:
har gått ut ifra dokumentasjonen av Amplify: https://aws-amplify.github.io/docs/js/authentication